lib/request.js

```diff
@@ -32,11 +33,12 @@ var MockRequest = function (options) {
 
   // setHeader now applies toLowerCase the names to mirror native node behaviour
   var self = this;
-  _.each(options.headers, function (v, k) {
-    if (v !== undefined) {
-      self.setHeader(k, v);
+
+  for(var k in options.headers) {
```

Comment:
This does have subtle side effects but I do prefer to remove the lodash dependency.